### PR TITLE
Hollow JAR limitations (SWARM-1916)

### PR DESCRIPTION
**Pull request by tradej**
Resolves SWARM-1916

- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: docs/concepts/packaging-types.adoc [hunk 1]
```diff
@@ -71,3 +71,23 @@ The hollow JARs are available under the following coordinates:
...
 
+[NOTE]
+====
+anchor:hollow-jar-limitations[]Using pre-built hollow JARs has certain limitations:
+
+* To enable WildFly Swarm to autodetect a JDBC driver, you must add the JAR with the driver to the `swarm.classpath` system property, for example:
++
+[source,bash,options="nowrap"]
+----
+$ java -Dswarm.classpath=./h2-1.4.196.jar -jar microprofile-jpa-hollow-swarm.jar myApp.war
```
Comment (tradej) [inline docs/concepts/packaging-types.adoc]: Not sure if the command line is correct for this use case; I have used the one from #835.

File: docs/concepts/packaging-types.adoc [hunk 2]
```diff
@@ -71,3 +71,23 @@ The hollow JARs are available under the following coordinates:
...
+----
+
+* YAML configuration files in your application that are named `project-*.yml` are not automatically applied.
+You must specify them manually, for example:
++
+[source,bash,options="nowrap"]
+----
+$ java -s project-myconfig.yml -jar microprofile-jpa-hollow-swarm.jar myApp.war
```
Comment (tradej) [inline docs/concepts/packaging-types.adoc]: Again, please check if it's correct for this use case.

File: docs/concepts/packaging-types.adoc [hunk 3]
```diff
@@ -71,3 +71,23 @@ The hollow JARs are available under the following coordinates:
 </dependency>
 ----
 
+[NOTE]
+====
+anchor:hollow-jar-limitations[]Using pre-built hollow JARs has certain limitations:
```
Comment (Ladicek) [inline docs/concepts/packaging-types.adoc]: These are general limitations of all hollow JARs, not just the pre-built hollow JARs.

File: docs/concepts/packaging-types.adoc [hunk 4]
```diff
@@ -71,3 +71,23 @@ The hollow JARs are available under the following coordinates:
...
+----
+$ java -Dswarm.classpath=./h2-1.4.196.jar -jar my-hollow-swarm.jar myApp.war
+----
+
+* YAML configuration files in your application are not automatically applied.
+You must specify them manually, for example:
++
+[source,bash,options="nowrap"]
+----
+$ java -jar microprofile-jpa-hollow-swarm.jar myApp.war -s ./project-defaults.yml
```
Comment (Ladicek) [inline docs/concepts/packaging-types.adoc]: `my-hollow-swarm.jar` here too, please; otherwise LGTM.

**Comment by heiko-braun** [comment]
@tradej what's that status of this? Can it be merged?

**Comment by tradej** [comment]
AFAICT yes.

**Review by Ladicek — APPROVED** [review]

**Pull request merged**